### DISCUSSION TITLE
Hi. It keep sleeping at one specific tweet and repeating sleeping...

### DISCUSSION THREAD

**Discussion by Geneseo** [tweepy/tweepy#1439] [Questions] [answered]
I have 500 Tweet ids and try to collect 3200 recent tweets from each tweet_id. The below is my code. but the problem is that it doesn't move on but stuck at some tweet that I am trying to collect.. On my termial, It keep iterating and shows me that 
"
getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 Rate limit reached. Sleeping for: 600 getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 ...1545 tweets downloaded so far getting tweets before 1117195540448432127 Rate limit reached. Sleeping for: 644 ''' 
So I think there must be exception occurered when I am trying to collect tweet from user: **halfbare**..? but I really don't know how to handle this... Can Anyone help me please?
"

My code is like as below:
`
import tweepy
import csv

def get_all_tweets(screen_name):

    consumer_key = *
    consumer_secret = * 
    access_key = *
    access_secret = *

    #authorize twitter, initialize tweepy
    auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
    auth.set_access_token(access_key, access_secret)
    api = tweepy.API(auth, wait_on_rate_limit=True, wait_on_rate_limit_notify=True)

    #initialize a list to hold all the tweepy Tweets & list with no retweets
    alltweets = []
    noRT = []

    #make initial request for most recent tweets with extended mode enabled to get full tweets
    new_tweets = api.user_timeline(screen_name = screen_name, tweet_mode = 'extended', count=200, include_retweets=False)

    #save most recent tweets
    alltweets.extend(new_tweets)

    #save the id of the oldest tweet less one
    oldest = alltweets[-1].id - 1

    #keep grabbing tweets until the api limit is reached
    while len(alltweets) <= 3200:
        print("getting tweets before {}".format(oldest))

        #all subsiquent requests use the max_id param to prevent duplicates
        new_tweets = api.user_timeline(screen_name = screen_name,tweet_mode = 'extended', count=200,max_id=oldest, include_retweets=False)

        #save most recent tweets
        alltweets.extend(new_tweets)

        #update the id of the oldest tweet less one
        oldest = alltweets[-1].id - 1

        print("...{} tweets downloaded so far".format(len(alltweets)))

        #removes retweets
    for tweet in alltweets:
        if 'RT' in tweet.full_text:
            continue
        else:
            noRT.append([tweet.id_str, tweet.created_at, tweet.full_text.encode("utf-8")])

            #, tweet.user, tweet.coordinates, tweet.place

    #write to csv
    with open('{}_tweets.csv'.format(screen_name), 'w') as f:
        writer = csv.writer(f)
        writer.writerow(["id","created_at","text"])
        writer.writerows(noRT)
        print('{}_tweets.csv was successfully created.'.format(screen_name))
    pass

if __name__ == '__main__':
        #pass in the username of the account you want to download
        usernames = ["halfbare","Huw_Merlin"]
        #"Fla_Medic","GertvanderHoek",
        for x in usernames:
            get_all_tweets(x)
`

**Comment by Harmon758** [comment, marked as answer]
https://twitter.com/halfbare seems to only have around that many Tweets:
```py
>>> user = api.get_user("halfbare")
>>> user.statuses_count
1550
```
![image](https://user-images.githubusercontent.com/9403740/93692908-8cbabd00-fabe-11ea-880d-2d6c3f709296.png)

Once your code reaches their earliest Tweet (which seems to be https://twitter.com/halfbare/status/1117195540448432128), your call to `API.user_timeline` will return an empty list since there are no older Tweets. Therefore, `alltweets` remains the same size, under 3200, and your while loop loops infinitely and spams requests to Twitter's GET statuses/user_timeline endpoint.

You should look into using a Cursor object to paginate instead.
See https://tweepy.readthedocs.io/en/latest/cursor_tutorial.html.

For code block usage, see https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks.
